bindings/experimental/distrdf/python/DistRDF/Profiling/Base.py

```diff
@@ -0,0 +1,27 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+class ProfilingData(ABC):
+    """
+    Base class for the collection of any type of profiling data
+    """
+
+    @abstractmethod
+    def Merge(self, other:"ProfilingData"):
+        """
+        Profiled data must provide a method to merge two instances of the same type.
+        """
+        raise NotImplementedError
+
+class Visualization:
+    """
+    Base class that represents visualizations for profiling data
+    """
+
+    @property
+    def Decorator(self) -> Callable:
+        return lambda func: func
+
+    @property
+    def Client_task(self) -> Callable:
+        return lambda task_result: None
```

Comment:
These two properties are probably better suited as abstract methods? At least you wouldn't have to define them in this class where they don't actually do anything